catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/KeyValueParser.java

```diff
@@ -0,0 +1,96 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class used for parsing key-value pairs in the format of <i>'key = value'</i>. Uses a regex
+ * under the hood. In plain English, the rule states that any {@link String} is a valid key-value
+ * pair if, for key {@code K} and for value {@code V}, the representation is {@code K=V} where the
+ * alphabet for {@code K} and {@code V} is <i>any</i> symbol except {@code =}.
+ * <p>
+ * Note that neither {@code K} or {@code V} can be empty. Any whitespace leading or tailing the key
+ * and value after parsing is removed. Whitespace <i>within</i> the key or value is preserved, but
+ * a key or value cannot consist entirely of whitespace.
+ */
+public class KeyValueParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeyValueParser.class);
+
+    private static final String KEY_VALUE_REGEX = "[^=]+[=][^=]+";
+
+    private static final String SPLIT_REGEX = "=";
+
+    private static final int SPLIT_LIMIT = 2;
+
+    private final boolean failFast;
+
+    /**
+     * Setup a default instance of {@link KeyValueParser}.
+     */
+    public KeyValueParser() {
+        this.failFast = false;
+    }
+
+    /**
+     * Sets up an instance of {@link KeyValueParser} that throws an {@link IllegalArgumentException}
+     * on the first failure during batch operations.
+     *
+     * @param failFast Parsing failures throw exceptions if true; otherwise they are silently ignored.
+     */
+    public KeyValueParser(final boolean failFast) {
+        this.failFast = failFast;
+    }
+
+    /**
+     * Given a string, returns true if it is a valid key-value pair. This method is useful for
+     * external policy validation.
+     *
+     * @param pair The string suspected to be of the form "key=value"
+     * @return {@code True} if the input is a valid key-value pair. False otherwise.
+     */
+    public boolean validatePair(final String pair) {
+        return pair.trim()
+                .matches(KEY_VALUE_REGEX);
+    }
+
+    /**
+     * Transforms a {@link List} of strings representing key-value pairs into a {@link Map} with
+     * only the valid mappings.
+     *
+     * @param pairList The list of strings of the form "key=value".
+     * @return A map of key -> value for only the validated key-value strings in the given list.
+     */
+    public Map<String, String> parsePairsToMap(final List<String> pairList) {
+        Map<String, String> pairMap = new HashMap<>();
+        for (String pair : pairList) {
```

Comment:
Should we check to see if pairList is null before the loop so we avoid a potential NullPointerException?